### PR TITLE
[Theme] Covered Unit Test for \Magento\Theme\Controller\Result\JsFooterPlugin

### PR DESCRIPTION
**Pull request by srsathish92**
### Description (*)
[Theme] Covered Unit Test for \Magento\Theme\Controller\Result\JsFooterPlugin

### Manual testing scenarios (*)
`vendor/bin/phpunit -c dev/tests/unit/phpunit.xml.dist app/code/Magento/Theme/Test/Unit/Controller/Result/JsFooterPluginTest.php`

### Questions or comments
If any please let me know your feedback

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by VladimirZaets — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @VladimirZaets, thank you for the review.
[ENGCOM-6867](https://jira.corp.magento.com/browse/ENGCOM-6867) has been created to process this Pull Request
:eight_spoked_asterisk: @VladimirZaets, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by lbajsarowicz** [comment]
image
Could you cover that case, too?

**Comment by srsathish92** [comment]
> image
> Could you cover that case, too?

@lbajsarowicz yes, I have covered & updated, please review now.

File: app/code/Magento/Theme/Test/Unit/Controller/Result/JsFooterPluginTest.php [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+     */
+    public function ifGetContentIsNotAStringDataProvider(): array
+    {
+        return [
+            [
+                'content' => []
+            ],
+            [
+                'content' => NULL
+            ]
```
Comment (lbajsarowicz) [inline app/code/Magento/Theme/Test/Unit/Controller/Result/JsFooterPluginTest.php]: You can replace this with:
```php
'empty_array' => ['content' => []],
'null' => ['content' => NULL]
```

As a result - when one of them fail - You'll know on which data set it failed (key of dataProvider is added to error message (Test failed with: X))

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
Please add keys that description for DataProvider.
Then we need to check if everything works as expected.

File: app/code/Magento/Theme/Test/Unit/Controller/Result/JsFooterPluginTest.php (hunk 1 above)
Comment (srsathish92) [reply app/code/Magento/Theme/Test/Unit/Controller/Result/JsFooterPluginTest.php]: @lbajsarowicz Thanks for the feedback, I have updated the keys.

Comment (srsathish92) [reply app/code/Magento/Theme/Test/Unit/Controller/Result/JsFooterPluginTest.php]: @lbajsarowicz Thanks for the feedback, I have updated the keys.

**Review by lbajsarowicz — APPROVED** [review]
Tests passed, thanks for your contribution.

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-6867](https://jira.corp.magento.com/browse/ENGCOM-6867) has been created to process this Pull Request
:eight_spoked_asterisk: @lbajsarowicz, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Comment by engcom-Charlie** [comment]
Failed functional test not related to the changes in this PR test fails in 2.4-develop with MSI modules.
image

**Pull request merged**